sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_convert_to_dict.py

```diff
@@ -0,0 +1,89 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: sample_convert_to_dict.py
+
+DESCRIPTION:
+    This sample demonstrates how to convert a model returned from an analyze operation
+    to a JSON serializable dictionary.
+
+USAGE:
+    python sample_convert_to_dict.py
+
+    Set the environment variables with your own values before running the sample:
+    1) DOCUMENTINTELLIGENCE_ENDPOINT - the endpoint to your Form Recognizer resource.
+    2) DOCUMENTINTELLIGENCE_API_KEY - your Form Recognizer API key
+"""
+
+import os
+import json
+
+
+def convert_to_and_from_dict():
+    path_to_sample_documents = os.path.abspath(
+        os.path.join(
+            os.path.abspath(__file__),
+            "..",
+            "./sample_forms/forms/Form_1.jpg",
+        )
+    )
+
+    from azure.core.serialization import AzureJSONEncoder
+    from azure.core.credentials import AzureKeyCredential
+    from azure.ai.documentintelligence import DocumentIntelligenceClient
+    from azure.ai.documentintelligence.models import AnalyzeResult
+
+    endpoint = os.environ["DOCUMENTINTELLIGENCE_ENDPOINT"]
+    key = os.environ["DOCUMENTINTELLIGENCE_API_KEY"]
+
+    document_analysis_client = DocumentIntelligenceClient(
+        endpoint=endpoint, credential=AzureKeyCredential(key)
+    )
+    with open(path_to_sample_documents, "rb") as f:
+        poller = document_analysis_client.begin_analyze_document(
+            "prebuilt-document", document=f, content_type="application/octet-stream"
+        )
+    result: AnalyzeResult = poller.result()
+
+    # convert the received model to a dictionary
+    analyze_result_dict = result.as_dict()
+
+    # save the dictionary as JSON content in a JSON file, use the AzureJSONEncoder
+    # to help make types, such as dates, JSON serializable
+    # NOTE: AzureJSONEncoder is only available with azure.core>=1.18.0.
+    with open("data.json", "w") as output_file:
+        json.dump(analyze_result_dict, output_file, cls=AzureJSONEncoder)
```

Comment:
You shouldn't need this line, `as_dict` should give you something that can be passed directly to `json.dumps`

Reply:
I tried to pass as_dict directly to json.dumps, but the result is not what I would expect from a json file, no indentations and in general it is somewhat similar to how you would store a text file.

Where as json.dump has what a json file should like

Reply:
@muizzhaque  You should able to format the output with either `json.dumps` or `json.dump`, for example:
```python
with open("data.json", "w") as output_file:
    json.dump(analyze_result_dict, output_file, indent=4)
```
or
```python
json.dumps(analyze_result_dict, indent=4)
```